libr/util/list.c

```diff
@@ -414,33 +472,113 @@ R_API void *r_list_get_by_string(const RList *list, int off, const char *str) {
 	char *p;
 	RListIter *iter;
 	r_list_foreach (list, iter, p) {
-		const char *ptr = p+off;
-		if (!strcmp (str, ptr))
+		const char *ptr = p + off;
+		if (!strcmp (str, ptr)) {
 			return p;
+		}
 	}
 	return NULL;
 }
 
-R_API RListIter *r_list_contains (const RList *list, const void *p) {
+R_API RListIter *r_list_contains(const RList *list, const void *p) {
 	void *q;
 	RListIter *iter;
 	r_list_foreach (list, iter, q) {
-		if (p == q)
+		if (p == q) {
 			return iter;
+		}
 	}
 	return NULL;
 }
 
-R_API RListIter *r_list_find (const RList *list, const void *p, RListComparator cmp) {
+R_API RListIter *r_list_find(const RList *list, const void *p, RListComparator cmp) {
 	void *q;
 	RListIter *iter;
 	r_list_foreach (list, iter, q) {
-		if (cmp (p, q) == 0)
+		if (!cmp (p, q)) {
 			return iter;
+		}
 	}
 	return NULL;
 }
 
+
+RListIter *_sortedmerge(RListIter *first, RListIter *second, RListComparator cmp) {
```

Comment:
I think it's probably better to make it iterative instead of recursive, also considering that it's not tail recursion so it can't be optimised.